thirtyinch/src/test/java/net/grandcentrix/thirtyinch/distinctuntilchanged/DistinctUntilChangedTest.java

```diff
@@ -0,0 +1,194 @@
+package net.grandcentrix.thirtyinch.distinctuntilchanged;
+
+import net.grandcentrix.thirtyinch.TiView;
+
+import org.junit.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsEqual.equalTo;
+
+/**
+ * Testing {@link DistinctUntilChanged} annotation via {@link DistinctUntilChangedInterceptor}
+ * requires interfaces extending {@link TiView}.
+ */
+public class DistinctUntilChangedTest {
+
+    private class CountWrapper {
+
+        private int mCalled = 0;
+
+        public void call() {
+            mCalled++;
+        }
+
+        public int getCalled() {
+            return mCalled;
+        }
+    }
+
+    /**
+     * This class doesn't have a good hashcode method
+     */
+    private class BadHashObject {
+
+        private final String mTest;
+
+        public BadHashObject(final String test) {
+            mTest = test;
+        }
+
+        @Override
+        public boolean equals(final Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+
+            final BadHashObject that = (BadHashObject) o;
+
+            return mTest != null ? mTest.equals(that.mTest) : that.mTest == null;
+
+        }
+
+        @Override
+        public int hashCode() {
+            return 42;
+        }
+    }
+
+    /**
+     * This class doesn't have a good equals method
+     */
+    private class BadEqualsObject {
+
+        private final String mTest;
+
+        public BadEqualsObject(final String test) {
+            mTest = test;
+        }
+
+        @Override
+        public int hashCode() {
+            return mTest != null ? mTest.hashCode() : 0;
+        }
+    }
+
+    private interface TestViewHash extends TiView {
+
+        @DistinctUntilChanged(comparator = HashComparator.class)
+        void annotatedMethod(String str);
+    }
+
+    private interface TestViewEquals extends TiView {
+
+        @DistinctUntilChanged(comparator = EqualsComparator.class)
+        void annotatedMethod(String str);
+    }
+
+    private interface TestViewBadHash extends TiView {
+
+        @DistinctUntilChanged(comparator = EqualsComparator.class)
+        void annotatedMethod(BadHashObject bho);
+    }
+
+    private interface TestViewBadEquals extends TiView {
+
+        @DistinctUntilChanged(comparator = HashComparator.class)

```

Comment:
Name and comparator isn't equal. You say `BadEquals` (and argument is `BadEquals`) but using `HashComparator `